tests/integration/flows/test_mysql_api_extended.py

```diff
@@ -0,0 +1,385 @@
+def setup_trigger_db():
+    """Function-scoped fixture to ensure a clean DB for each trigger test."""
+    db_name = f"trigger_test_db_{uuid.uuid4().hex[:8]}"
+    params = {"user": "postgres", "password": "postgres", "host": "postgres", "port": 5432, "database": "postgres"}
+    source_table_name = f"source_table_{uuid.uuid4().hex[:8]}"
+    target_table_name = f"target_table_{uuid.uuid4().hex[:8]}"
+    helper = BaseStuff()
+    helper.use_binary = False
+    try:
+        print(
+            f"\n--> [Fixture setup_trigger_db] Setting up local database: {db_name} on {params['host']}:{params['port']}"
+        )
+        helper.query(f"DROP DATABASE IF EXISTS {db_name}")
+        create_datasource_sql_via_connector(helper, db_name, "postgres", params)
+
+        helper.query(f"CREATE TABLE {db_name}.{source_table_name} (id INT, message VARCHAR(255));")
+        helper.query(f"CREATE TABLE {db_name}.{target_table_name} (id INT, message VARCHAR(255));")
+        helper.query(f"INSERT INTO {db_name}.{source_table_name} (id, message) VALUES (101, 'initial_update_message');")
+        helper.query(f"INSERT INTO {db_name}.{source_table_name} (id, message) VALUES (102, 'initial_delete_message');")
+        yield db_name, source_table_name, target_table_name
+    except mysql.connector.Error as setup_err:
+        pytest.fail(f"Trigger fixture setup failed. Ensure Docker environment is running. Error: {setup_err}")
+    finally:
+        print(f"\n--> [CLEANUP] Dropping DATABASE: {db_name}")
+        try:
+            helper.query(f"DROP DATABASE IF EXISTS {db_name};")
+        except mysql.connector.Error:
+            pass
+
+
+@pytest.mark.parametrize("use_binary", [False, True], indirect=True)
+class TestMySQLTriggers(BaseStuff):
+    """Test suite for Trigger operations."""
+
+    @pytest.fixture
+    def use_binary(self, request):
+        self.use_binary = request.param
+
+    @pytest.mark.usefixtures("setup_trigger_db")
+    def test_trigger_lifecycle_update(self, setup_trigger_db, use_binary):
+        db_name, source_table_name, target_table_name = setup_trigger_db
+        trigger_name = f"test_update_trigger_{uuid.uuid4().hex[:8]}"
+        test_id = 101
+        updated_message = "this message was updated"
+        try:
+            # Ensure the target table is empty before each test run.
+            self.query(f"DELETE FROM {db_name}.{target_table_name};")
+
+            create_trigger_query = f"""
+                CREATE TRIGGER {trigger_name}
+                ON {db_name}.{source_table_name}
+                (INSERT INTO {db_name}.{target_table_name} (id, message) SELECT id, message FROM TABLE_DELTA);
+            """
+            self.query(create_trigger_query)
+            time.sleep(5)  # Allow time for trigger creation
+
+            # Activate Trigger
+            self.query(f"UPDATE {db_name}.{source_table_name} SET message = '{updated_message}' WHERE id = {test_id};")
+            self.query("COMMIT;")
```

Comment:
just in case, transactions are not supported, the query above will be commited automatically